python/paddle/fluid/io.py

```diff
@@ -454,3 +452,149 @@ def get_parameter_value_by_name(name, executor, program=None):
         program = default_main_program()
     var = program.global_block().var(name)
     return get_parameter_value(var, executor)
+
+
+SUCCESS = "_SUCCESS"
+BEGIN_SECS = None
+
+
+def save_checkpoint(executor,
+                    dirname,
+                    keep_max=3,
+                    save_secs=600,
+                    main_program=None):
+    """
+    Save Variables to Checkpint Dir
+
+    :param dirname
+    :param keep_max
+    :param save_secs
+    :param main_program
+    """
+    if dirname is None:
+        raise Exception("save checkpoint dir can not be none")
+
+    if not os.path.isdir(dirname):
+        os.makedirs(dirname)
+
+    global BEGIN_SECS
+    if BEGIN_SECS is not None:
+        if time.time() - BEGIN_SECS < save_secs:
+            return
+    BEGIN_SECS = time.time()
+
+    serial = _get_lastest_checkpoint_dir(dirname) + 1
+    cur_dir = os.path.join(dirname, str(serial))
+    # save_persistables(executor, cur_dir, main_program)
+    save_vars(
```

Comment:
why call `save_vars` instead of `save_psersistables`?

Reply:
save_psersistables can not filter out gradient variables.